micro-rdk/src/native/entry.rs

```diff
@@ -233,7 +240,7 @@ pub fn serve_with_provisioning<S>(
     let mut last_error = None;
     let (app_config, tls_server_config) = loop {
         match cloned_exec.block_on(serve_provisioning_async(
-            ip,
+            network.get_ip(),
```

Comment:
Similar thoughts here as above in this file. In particular, this comment from `ExternallyManagedNetwork`:
```
    // TODO: provide a way for an external managed network to communicate a change in IP
    // address
```

When that happens, will we remember to revisit the code here?